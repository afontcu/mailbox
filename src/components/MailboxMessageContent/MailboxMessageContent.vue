<template>
  <main class="MailboxMessageContent">
    <div v-if="uid" class="MailboxMessageContent__wrapper">
      <h2>{{ subject }}</h2>
      <p class="MailboxMessageContent__message">{{ message }}</p>
      <div class="MailboxMessageContent__meta">
        <p>{{ sender }}</p>
        <p>{{ date }}</p>
      </div>
    </div>
    <p v-else class="MailboxMessageContent__no-message">
      Select a message to display its content.
    </p>
  </main>
</template>

<script>
export default {
  name: 'MailboxMessageContent',
  props: {
    uid: { type: String, default: '' },
    sender: { type: String, default: '' },
    subject: { type: String, default: '' },
    date: { type: String, default: '' },
    message: { type: String, default: '' },
  },
}
</script>

<style scoped>
:root {
  --content-shadow-color: rgba(63, 63, 68, 0.15);
}

.MailboxMessageContent {
  display: flex;
}

.MailboxMessageContent__no-message {
  flex: 1;
  align-self: center;
  text-align: center;
}

.MailboxMessageContent__wrapper {
  padding: var(--global-spacing-large);
  margin: 0;

  background-color: #fff;
  transition: all ease var(--global-transition);
  border-radius: var(--global-radius);
  box-shadow: 0 0 0 1px rgba(63, 63, 68, 0.05),
    0 1px 3px 0 var(--content-shadow-color);
}

.MailboxMessageContent__message {
  font-size: 1.2rem;
  line-height: 1.5;
}

.MailboxMessageContent__meta {
  display: flex;
  justify-content: space-between;
  padding-top: var(--global-spacing-large);
  margin-top: var(--global-spacing-large);

  border-top: 1px solid var(--content-shadow-color);
  font-weight: bold;
}
</style>
