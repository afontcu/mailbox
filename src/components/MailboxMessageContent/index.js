import MailboxMessageContent from './MailboxMessageContent'

export default MailboxMessageContent
