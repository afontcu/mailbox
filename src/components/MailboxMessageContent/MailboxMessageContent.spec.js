import MailboxMessageContent from './index.js'
import { mount } from '@vue/test-utils'

const propsData = {
  uid: '1',
  sender: 'sender',
  subject: 'subject',
  time_sent: 1456744407,
  message: 'lorem ipsum',
}

describe('MailboxMessageContent.vue', () => {
  it('is a component', () => {
    const wrapper = mount(MailboxMessageContent, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(MailboxMessageContent, { propsData })
    expect(wrapper.classes()).toContain('MailboxMessageContent')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(MailboxMessageContent, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
