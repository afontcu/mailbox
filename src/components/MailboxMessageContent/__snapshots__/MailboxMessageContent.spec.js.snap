// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailboxMessageContent.vue renders a snapshot: default 1`] = `
<main class="MailboxMessageContent" time_sent="1456744407">
  <div class="MailboxMessageContent__wrapper">
    <h2>subject</h2>
    <p class="MailboxMessageContent__message">lorem ipsum</p>
    <div class="MailboxMessageContent__meta">
      <p>sender</p>
      <p></p>
    </div>
  </div>
</main>
`;
